### PR TITLE
add AdaGrad optimizer in python, a revised version

### PR DESCRIPTION
**Pull request by linmx0130**
@winstywang @sxjscience Sorry for my bad code. Thanks for commenting. This version is a revised version.

### CODE REVIEW

File: python/mxnet/optimizer.py [hunk 1]
```diff
@@ -547,6 +547,44 @@ def update(self, index, weight, grad, state):
...
+        return zeros(weight.shape, weight.context)   #history
+    def update(self, index, weight, grad, state):
+        assert(isinstance(weight, NDArray))
+        assert(isinstance(grad, NDArray))
+        grad = grad*self.rescale_grad
+        history = state
+        history[:] += (grad * grad)
+        delta = zeros(weight.shape, weight.context)
+        delta[:] = -self.lr * (grad / sqrt(history + self.float_stable_eps) + self.wd*weight)
+        weight[:] += delta
```
Comment (sxjscience) [inline python/mxnet/optimizer.py]: Can we directly use `weight[:] += -self.lr * (grad / sqrt(history + self.float_stable_eps) + self.wd*weight)` ?

Comment (linmx0130) [inline python/mxnet/optimizer.py]: I have not tried but it should be correct.

Comment (sxjscience) [inline python/mxnet/optimizer.py]: Also, I think there should be a blank line after each member function and spaces for `*` OP.

**Comment by linmx0130** [comment]
Updated.

**Comment by piiswrong** [comment]
LGTM

**Pull request merged**